### PR TITLE
boards: arm: mimxrt1020_evk: SD Card

### PR DESCRIPTION
**Pull request by JakeMercer**
Porting code from mimxrt1050_evk to this board to enable use of the SD
card.

Tested with samples/subsys/fs/fat_fs and a modification to its prj.conf
to enable the CONFIG_DISK_DRIVER_SDMMC

Signed-off-by: Jake Mercer <jake.mercer@civica.co.uk>

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Looks great, thanks for the contribution!

Will you please also update `boards/arm/mimxrt1020_evk/doc/index.rst` to reflect this change?

No modification to `samples/subsys/fs/fat_fs/prj.conf` is necessary since you've configured `DISK_DRIVER_SDMMC` in `boards/arm/mimxrt1020_evk/Kconfig.defconfig`

**Comment by JakeMercer** [comment]
How does that look? :)

**Review by MaureenHelm — APPROVED** [review]
> How does that look? :)

Looks great, thank you! The merge window is currently closed while we stabilize for the 2.6 release, so we need to wait until that is complete (planned May 28) before we can merge this.

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**
